provider/maas/maas2instance.go

```diff
@@ -65,10 +71,19 @@ func (mi *maas2Instance) Addresses() ([]network.Address, error) {
 // Status returns a juju status based on the maas instance returned
 // status message.
 func (mi *maas2Instance) Status() instance.InstanceStatus {
-	// TODO (babbageclunk): this should rerequest to get live status.
-	statusName := mi.machine.StatusName()
-	statusMsg := mi.machine.StatusMessage()
+	args := gomaasapi.MachinesArgs{SystemIDs: []string{mi.machine.SystemID()}}
+	machines, err := mi.maasController.Machines(args)
+	if err != nil {
```

Comment:
> Are you sugesting we change the interface?

Ah, I forgot that the interface didn't return an error. The way all the existing code works is: to get fresh status, you get a new Instance.

So I think you should revert this function to how it was, drop the TODO, and update pollInstanceStatus to get a fresh Instance each time around the loop. Alternatively, don't call Status(), but call some internal function. Getting a fresh Instance is probably simplest.